swift/example_code/sqs/scenario/Sources/entry.swift

```diff
@@ -0,0 +1,716 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+//
+// This example demonstrates how to use Amazon Simple Notification Service and
+// Amazon Simple Queue Service together to create queues and publish messages
+// to them through a single topic. The example demonstrates various features
+// of both SNS and SQS together.
+
+import ArgumentParser
+import AWSClientRuntime
+import AWSSNS
+import AWSSQS
+import Foundation
+
+struct ExampleCommand: ParsableCommand {
+    @Option(help: "Name of the Amazon Region to use")
+    var region = "us-east-1"
+
+    static var configuration = CommandConfiguration(
+        commandName: "queue-scenario",
+        abstract: """
+        This example interactively demonstrates how to use Amazon Simple
+        Notification Service (Amazon SNS) and Amazon Simple Queue Service
+        (Amazon SQS) together to publish and receive messages using queues.
+        """,
+        discussion: """
+        Supports filtering using a "tone" attribute.
+        """
+    )
+
+    /// Prompt for an input string. Only non-empty strings are allowed.
+    /// 
+    /// - Parameter prompt: The prompt to display.
+    ///
+    /// - Returns: The string input by the user.
+    func stringRequest(prompt: String) -> String {
+        var str: String?
+
+        while str == nil {
+            print(prompt, terminator: "")
+            str = readLine()
+
+            if str != nil && str?.count == 0 {
+                str = nil
+            }
+        }
+
+        return str!
+    }
+
+    /// Ask a yes/no question.
+    /// 
+    /// - Parameter prompt: A prompt string to print.
+    ///
+    /// - Returns: `true` if the user answered "Y", otherwise `false`.
+    func yesNoRequest(prompt: String) -> Bool {
+        var answer: String?
+
+        while answer == nil {
+            answer = stringRequest(prompt: prompt)
+
+            if answer != nil {
+                answer = answer!.lowercased()
+
+                if answer != "y" && answer != "n" {
+                    print("Please answer 'Y' or 'N'. ", terminator: "")
+                    answer = nil
+                }
+            }
+        }
+        return answer == "y"
+    }
+
+    /// Display a menu of options then request a selection.
+    /// 
+    /// - Parameters:
+    ///   - prompt: A prompt string to display before the menu.
+    ///   - options: An array of strings giving the menu options.
+    ///
+    /// - Returns: The index number of the selected option or 0 if no item was
+    ///   selected.
+    func menuRequest(prompt: String, options: [String]) -> Int {
+        let numOptions = options.count
+
+        if numOptions == 0 {
+            return 0
+        }
+
+        print(prompt)
+
+        var index = 0
+        for option in options {
+            print("(\(index)) \(option)")
+            index += 1
+        }
```

Comment:
Maybe use `.enumerated()` to get indices for options like you did below?

Reply:
Good catch.